src/test/java/com/brandwatch/internship/fetchandstoretweets/integrationtests/QueryIT.java

```diff
@@ -0,0 +1,66 @@
+package com.brandwatch.internship.fetchandstoretweets.integrationtests;
+
+import com.brandwatch.internship.fetchandstoretweets.controllers.QueryController;
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@TestPropertySource(locations="classpath:test.properties")
+public class QueryIT {
+
+    private static final long QUERY_ID = 3;
+    private static final String SEARCH_STRING = "test";
+
+    @Autowired
+    private QueryController queryController;
+
+    @Autowired
+    private QueryRepository queryRepository;
+
+
+    @Test
+    public void getAllQueriesTest() {
+        assertEquals(queryController.getAllQueries(), queryRepository.findAll());
+    }
+
+    @Test
+    public void getQueryByIdTest() {
+        try {
+            assertEquals(queryController.getQueryById(QUERY_ID), queryRepository.findOneById(QUERY_ID));
```

Comment:
It is a mistake. I modified it. The reason why I made it like this the first place was because I wanted to check the ids of the queries. If there were no query with that I did then a null pointer exception would be raised because of null.getId(). Actually after having a closer look, NoSuchQueryException will be raised first then a null pointer exception. Since I am comparing the queries now not their ids, I removed the second assert.